phpBB/phpbb/storage/storage.php

```diff
@@ -106,7 +115,28 @@ protected function get_adapter()
 	 */
 	public function put_contents($path, $content)
 	{
-		$this->get_adapter()->put_contents($path, $content);
+		$adapter = $this->get_adapter();
+		$storage = $this->get_name();
+
+		/**
+		* Before writing in the file
+		*
+		* @event core.storage_put_contents_before
```

Comment:
you can use multiple dots if you want (`core.storage.put_contents_before` would make sense per example)

Reply:
None of the events use multiple dots

Reply:
well, both are okayish